grails-app/domain/org/pih/warehouse/product/ProductSupplier.groovy

```diff
@@ -151,4 +151,21 @@ class ProductSupplier implements Serializable, Comparable<ProductSupplier> {
             "Date created"                        : ["property": "dateCreated", "dateFormat": "MM/dd/yyyy"],
             "Last updated"                        : ["property": "lastUpdated", "dateFormat": "MM/dd/yyyy"]
     ]
+
+    ProductSupplierListDto toJson() {
```

Comment:
As we've discussed in the past, I do not want to add a DTO layer unless there's a requirement to do so. Revert and make this the default toJson() method for ProductSupplier unless there's a compelling reason to add a DTO here.

Reply:
We should either follow the existing conventions or spend time on v2. 

REST Resource https://docs.grails.org/latest/guide/REST.html
JSON Views https://views.grails.org/latest/

However, since we haven't spiked Grails 3 REST support I would recommend following our existing conventions.

* Add Map toJson() to domain
* Add marshaller definition to Bootstrap 
* Use BaseDomainApiController unless there's a need to override REST endpoints
* List and search should be implemented using separate controller actions

Reply:
I expected you to comment on that one - the reason I did this was just to use types further in the method, not to use "defs" - as you can see, by doing so, I could easily say, what type will be returned from this method:
```groovy
List<ProductSupplierListDto> getProductSuppliers(...)
```
I know I could just return the `List<ProductSupplier>` and to the `toJson()` parsing in the controller, but I thought this is a more convenient way to parse entity to POJO (in the service layer). 
It was kind of instinct to do it this way, because if in Spring I wanted to do it this way, in the controller I'd already get the `LazyInitializationException` (because the transaction from the service commited). I don't know why it doesn't happen here, if we eventually have the osiv disabled. 

Of course I can revert that, but just wanted to explain, why I added the dto here - in a nutshell - not to do the `toJson()` parsing in the controller and not to make the returned type as `def`.

Reply:
I think we should investigate why we don't get `LazyInitializationException` here. I am pretty sure that we should get this error here. Why do you think @jmiranda?

Reply:
@alannadolny to be clear - I didn't mean we should get the lazy inside the `toJson()` now - I meant that we should expect to get it when trying to do the `.toJson()` in the controller layer (because we are not anymore inside a transaction boundary).

Reply:
Yeah, I meant doing `.toJson()` in the controllers

Reply:
For now I'd like to go with the convention we've been using with the domain providing the toJson() implementation and just returning a map. I totally get what you want to do but I don't want to litter the code with DTOs if we end up refactoring and using JSON views. If you want to try to go down the JSON view path (or whatever Grails has annointed the standard), then give it a shot. Otherwise, just do things the way we have been doing them until we find time to implement it in a more standardized way.

Reply:
As for the LazyInitializationException ... I suspose you're both right and I'm not exactly sure why we're not seeing the issue. Try to implement the toJson in the domain and have the toJson dig deep into an association that is not loaded in the criteria query.